featuretools/computational_backends/feature_set.py

```diff
@@ -204,8 +206,8 @@ def _dependent_uses_full_entity(self, feature):
 # These functions are used for sorting and grouping features
 
 
-def _get_use_previous(f):
-    if isinstance(f, AggregationFeature) and f.use_previous is not None:
+def _get_use_previous(f):  # TODO Sort and group features for DateOffset
```

Comment:
Can you update the TODO saying that its only necessary for DateOffset with two different temporal values?And also check if the DateOffset contains a unit before ignoring it?